screens/ActivitiesScreen.tsx

```diff
@@ -1,117 +1,159 @@
+      setLocation(locationToGive)
+    })()
+  }, [])
+
   return (
     <>
-      <EventsMap
-        ref={appRef}
-        style={{ width: "100%", height: "100%" }}
-        initialRegion={state.initialRegion}
-        renderMarker={(item) => (
-          <>
-            {/* Solid white background for the image */}
-            <ImageBackground
-              source={require("../assets/Solid_white.svg.png")}
-              style={{
-                flex: 1,
-                width: MARKER_SIZE,
-                height: MARKER_SIZE,
-                borderRadius: MARKER_SIZE / 2,
-                alignItems: "center",
-                justifyContent: "center",
-                overflow: "hidden"
-              }}
-            />
-            {/* The profile image itself */}
-            <ImageBackground
-              source={require("../assets/icon.png")}
-              style={{
-                marginTop: "3%",
-                position: "absolute",
-                width: MARKER_SIZE - 5,
-                height: MARKER_SIZE - 5,
-                borderRadius: MARKER_SIZE - 5 / 2,
-                alignSelf: "center",
-                overflow: "hidden"
-              }}
-            />
-            {/* Ideally, badge showing current number of attendees */}
-            <View
-              style={{
-                marginLeft: "30%",
-                paddingHorizontal: 3,
-                backgroundColor: "red",
-                position: "absolute",
-                borderRadius: 10,
-                flexDirection: "row",
-                justifyContent: "space-between",
-                alignItems: "center"
-              }}
-            >
-              <Icon name="people" type="ionicon" color="white" />
-              <Text
-                style={{
-                  paddingHorizontal: 4,
-                  fontWeight: "bold",
-                  fontSize: 16,
-                  color: "white"
-                }}
-              >
-                {/* Ideally, add the event amount of attendees here */}
-                {5}
-              </Text>
-            </View>
-          </>
+      {location
+        ? (
+          <EventsMap
+            ref={appRef}
+            style={{ width: "100%", height: "100%" }}
+            initialRegion={{
+              latitude: location.coords.latitude,
+              longitude: location.coords.longitude,
+              latitudeDelta: 0.5,
+              longitudeDelta: 0.5
+            }}
+            renderMarker={(item) => (
+              <>
+                <View
+                  style={[
+                    styles.badgeDisplay,
+                    {
+                      backgroundColor: events.find((x) => x.id === item.id).color
+                    }
+                  ]}
+                >
+                  <Ionicon name="people" color="white" size={10} />
+                  <Text style={styles.badgeText}>
+                    {events.find((x) => x.id === item.id).attendeeCount}
```

Comment:
Replacing with item.### instead of trying to verify? Iffy atm
Go to item in parentheses and put event type instead